Automattic/wordpress-activitypub
# Fix #493

Fixes #493

## Proposed changes:

* Replaces call to `wp_filter_kses( $data )` with `addslashes( wp_kses( $data, 'pre_comment_content' ) )` when creating a new comment. This also fixes disappearing slashes (the AP JSON content is not pre-slashed).
* Additionaly, this PR adds a `wp_kses_allowed_html` filter that allows comments to contain `p` and `br` tags (only for AP comments, as the filter is removed immediately after the content is inserted).

### Other information:

- I have not changed (or run) test cases.
- I believe WP core will _again_ run new comments created thru `wp_new_comment()` through a kses filter. It normally uses the `pre_comment_content` for this, so the tags will be allowed there, too (the filter is only removed after this second call).

## Testing instructions:






* Reply to a federated post (e.g., on Mastodon), and use newline characters. The resulting WP comment should now include the `p` or `br` elements that were stripped before.
* Do the same thing, but use backslashes (e.g., respond using `¯\_(ツ)_/¯`). The backslash that would previously be stripped should appear in the comment.
